Neutrino lint results differ from neutrino build

As part of upgrading to Neutrino v6, I was going through Treeherder's existing eslint `globals` configuration, to see if there were any entries that were no longer required. I found that if I commented out the `'React'` global, then whilst `neutrino lint` would report no errors, `neutrino build` returned:

```
× Building project failed
./ui/js/reactrevisions.jsx
Module build failed: Module failed because of a eslint error.

C:\Users\Ed\src\treeherder\ui\js\reactrevisions.jsx
  14:11  error  'React' is not defined  no-undef
  57:15  error  'React' is not defined  no-undef
  ...
```

At first I thought this might be a `NODE_ENV` difference, however it turns out to actually be due to the file being a `.jsx` file.

Looking at the source for [eslint-loader](https://github.com/MoOx/eslint-loader/blob/756330c19d2779c4102d40fb7bfbb1ad183b714b/index.js#L217) (which is what runs when using `neutrino build`) I see it uses eslint's `executeOnText()`, which just operates on the passed text regardless of file extension (since the webpack loader config already filtered by file extension).

However the `neutrino lint` command instead uses the eslint API via `executeOnFiles()` - and whilst that's likely the better choice for this use-case, means that if `include` is a directory (which it will be for Neutrino, since it will be set to `neutrino.options.source`), then eslint filters the files by the value for `extension`, which defaults to `['.js']`:
https://eslint.org/docs/developer-guide/nodejs-api#cliengine

As such, the lint command needs to override `extensions`, setting it to `['.js', '.jsx']`.

I'll open a PR.

--- a/docs/middleware/neutrino-middleware-eslint/README.md
+++ b/docs/middleware/neutrino-middleware-eslint/README.md
@@ -46,6 +46,7 @@ neutrino.use(eslint, {
     cwd: neutrino.options.root,
     useEslintrc: false,
     root: true,
+    extensions: ['js', 'jsx'],
     plugins: ['babel'],
     baseConfig: {},
     envs: ['es6'],

--- a/packages/neutrino-middleware-eslint/README.md
+++ b/packages/neutrino-middleware-eslint/README.md
@@ -46,6 +46,7 @@ neutrino.use(eslint, {
     cwd: neutrino.options.root,
     useEslintrc: false,
     root: true,
+    extensions: ['js', 'jsx'],
     plugins: ['babel'],
     baseConfig: {},
     envs: ['es6'],

--- a/packages/neutrino-middleware-eslint/index.js
+++ b/packages/neutrino-middleware-eslint/index.js
@@ -56,6 +56,9 @@ module.exports = (neutrino, opts = {}) => {
             cwd: neutrino.options.root,
             useEslintrc: false,
             root: true,
+            // eslint-loader uses executeOnText(), which ignores the `extensions` setting.
+            // However it's still needed for the lint command, as it uses executeOnFiles().
+            extensions: ['js', 'jsx'],
             plugins: ['babel'],
             baseConfig: {},
             envs: ['es6'],

